datafusion/core/tests/sql/union.rs

```diff
@@ -80,6 +80,23 @@ async fn union_all_with_aggregate() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn union_all_with_count() -> Result<()> {
+    let ctx = SessionContext::new();
+    execute_to_batches(&ctx, "CREATE table t as SELECT 1 as a").await;
+    let sql = "SELECT COUNT(*) FROM (SELECT a from t UNION ALL SELECT a from t)";
```

Comment:
👍  agrees with postgres:

```sql
postgres=# CREATE table t as SELECT 1 as a;
SELECT 1
postgres=# SELECT COUNT(*) FROM (SELECT a from t UNION ALL SELECT a from t);
ERROR:  subquery in FROM must have an alias
LINE 1: SELECT COUNT(*) FROM (SELECT a from t UNION ALL SELECT a fro...
                             ^
HINT:  For example, FROM (SELECT ...) [AS] foo.
postgres=# SELECT COUNT(*) FROM (SELECT a from t UNION ALL SELECT a from t) as sq;
 count 
-------
     2
(1 row)
```

Reply:
BTW, I'm not sure if it was a bug, but the `PushDownProjection` doesn't push down the empty projection to the child plans of `Union` in this example. Maybe @jackwener knows more about it.

Reply:
> BTW, I'm not sure if it was a bug, but the `PushDownProjection` doesn't push down the empty projection to the child plans of `Union` in this example. Maybe @jackwener knows more about it.

we need an issue to track this problem, I think there are some different ways of dealing with this problem.

Reply:
Could you please file an issue @jackwener ?